doc/options.md

```diff
@@ -0,0 +1,2865 @@
+# Options Reference 
+
+1. [`inputs.iogx.lib.mkFlake`](#mkflake) 
+    Makes the final flake outputs.
+2. [`pkgs.lib.iogx.mkProject`](#mkproject) 
+    Makes a [`haskell.nix`](https://github.com/input-output-hk/haskell.nix) project decorated with the `iogx` overlay.
+3. [`pkgs.lib.iogx.mkShell`](#mkshell)
+    Makes a `devShell` with `pre-commit-check` and tools.
+
+---
+
+### `mkFlake`
+
+**Type**: core API function
+
+
+
+**Example**: 
+```nix
+# flake.nix
+{
+  outputs = inputs: inputs.iogx.lib.mkFlake {
+    inherit inputs;
+    repoRoot = ./.;
+    debug = false;
+    nixpkgsArgs = {};
+    systems = [ "x86_64-linux" "x86_64-darwin" ];
+    outputs = { repoRoot, inputs, pkgs, lib, system }: [];
+  };
+}
+
+```
+
+
+The `inputs.iogx.lib.mkFlake` function takes an attrset of options and returns an attrset of flake outputs.
+
+In this document, options for the input attrset are prefixed by `mkFlake.<in>`.
+
+
+---
+
+### `mkFlake.<in>.debug`
+
+**Type**: boolean
+
+**Default**: `false`
+
+
+
+
+If enabled, IOGX will trace debugging info to standard output.
+
+
+---
+
+### `mkFlake.<in>.flake`
+
+**Type**: attribute set
+
+**Default**: `{ }`
+
+
+**Example**: 
+```nix
+{
+  lib.bar = _: null;
+
+  packages.x86_64-linux.foo = null;
+  devShells.x86_64-darwin.bar = null;
+
+  networks = {
+    prod = { };
+    dev = { };
+  };
+}
+
+```
+
+
+A flake-like attrset.
+
+You can place additional flake outputs here, which will be recursively updated with the attrset from [`mkFlake.<in>.outputs`](#mkflakeinoutputs).
+
+This is a good place to put system-independent values like a `lib` attrset or pure Nix values.
+
+
+---
+
+### `mkFlake.<in>.inputs`
+
+**Type**: attribute set
+
+
+
+
+
+Your flake inputs.
+
+You almost certainly want to do `inherit inputs;` here (see the example in [`mkFlake`](#mkflake))
+
+
+---
+
+### `mkFlake.<in>.nixpkgsArgs`
+
+**Type**: attribute set
+
+**Default**: 
+```nix
+{ 
+  config = { }; 
+  overlays = [ ]; 
+}
+
+```
+
+
+**Example**: 
+```nix
+# flake.nix
+{
+  outputs = inputs: inputs.iogx.lib.mkFlake {
+    inherit inputs;
+    outputs = import ./nix/outputs.nix;
+    nixpkgsArgs = {
+      overlays = [(self: super: { 
+        acme = super.callPackage ./nix/acme.nix { }; 
+      })];
+    };  
+  };
+}
+
+```
+
+
+Internally, IOGX calls `import inputs.nixpkgs {}` for each of your [`mkFlake.<in>.systems`](#mkflakeinsystems).
+
+Using `nixpkgsArgs` you can provide an additional `config` attrset and a list of `overlays` to be appended to nixpkgs.
+
+
+---
+
+### `mkFlake.<in>.outputs`
```

Comment:
I expect for most users this setting will be an opaque file of copy-pasted magic. What should I put here? Why can't it be done for me by default? I think this makes sense in an "advanced" interface, where the user might be doing complex things to generate the meat of the flake but then wants to use iogx to put it all together, but even I don't know what the right "default" setting is here or how to be confident that I've set everything I need.